packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -76,24 +92,20 @@ export class Checkbox extends BaseComponent<ICheckboxProps, ICheckboxState> impl
           name={ name || this._id }
           className={ css('ms-Checkbox-input', styles.input) }
           type='checkbox'
-          onChange={ this._onChange }
-          onFocus={ this._onFocus }
-          onBlur={ this._onBlur }
+          onChange={ this._onChange.bind(this) }
```

Comment:
don't use `.bind` in a react component. You're re-binding on every render call and passing in a mutating value each time, preventing it from ever becoming a pure component.

Instead, use the `@autobind` decorator on the callback. This will "bind" the method once on class instantiation.